components/encoder/fake/encoder_test.go

```diff
@@ -43,42 +49,44 @@ func TestEncoder(t *testing.T) {
 	t.Run("specify a type", func(t *testing.T) {
 		testutils.WaitForAssertion(t, func(tb testing.TB) {
 			tb.Helper()
-			ticks, positionType, err := e.GetPosition(context.Background(), pb.PositionType_POSITION_TYPE_TICKS_COUNT.Enum(), nil)
+			ticks, positionType, err := e.GetPosition(context.Background(), encoder.PositionType_POSITION_TYPE_TICKS_COUNT.Enum(), nil)
 			test.That(tb, err, test.ShouldBeNil)
 			test.That(tb, ticks, test.ShouldEqual, 0)
-			test.That(tb, positionType, test.ShouldEqual, pb.PositionType_POSITION_TYPE_UNSPECIFIED)
+			test.That(tb, positionType, test.ShouldEqual, encoder.PositionType_POSITION_TYPE_TICKS_COUNT)
 		})
 	})
 	t.Run("get properties", func(t *testing.T) {
 		testutils.WaitForAssertion(t, func(tb testing.TB) {
 			tb.Helper()
 			props, err := e.GetProperties(ctx, nil)
 			test.That(tb, err, test.ShouldBeNil)
-			test.That(tb, props[encoder.TicksCountSupported], test.ShouldBeFalse)
+			test.That(tb, props[encoder.TicksCountSupported], test.ShouldBeTrue)
 			test.That(tb, props[encoder.AngleDegreesSupported], test.ShouldBeFalse)
 		})
 	})
 
 	// Set Speed
 	t.Run("set speed", func(t *testing.T) {
-		err := e.SetSpeed(ctx, 1)
+		e1 := e.(*Encoder)
+		err := e1.SetSpeed(ctx, 1)
 		test.That(t, err, test.ShouldBeNil)
-		test.That(t, e.speed, test.ShouldEqual, 1)
+		test.That(t, e1.speed, test.ShouldEqual, 1)
 	})
 
 	// Start with default update rate
 	t.Run("start default update rate", func(t *testing.T) {
-		err := e.SetSpeed(ctx, 0)
+		e1 := e.(*Encoder)
+		err := e1.SetSpeed(ctx, 0)
 		test.That(t, err, test.ShouldBeNil)
 
-		e.Start(ctx)
+		// e1.Start(ctx)
```

Comment:
[nit] commented out, test still works?

Reply:
yeah now that there's a new function for the fake encoder it's started there

Reply:
just remove comment then, thanks!